cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_adapters.go

```diff
@@ -308,3 +312,75 @@ func (s *Server) getInstalledPackageDetail(pkgInstall *packagingv1alpha1.Package
 
 	return installedPackageDetail, nil
 }
+
+func (s *Server) newSecret(installedPackageName, values, targetNamespace string) (*k8scorev1.Secret, error) {
+	return &k8scorev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       k8scorev1.ResourceSecrets.String(),
+			APIVersion: k8scorev1.SchemeGroupVersion.WithResource(k8scorev1.ResourceSecrets.String()).String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			// TODO(agamez): think about name collisions
```

Comment:
Haha - there you go :P. Yes, I guess either way (ie. even if we clean up secrets on failed installs) we'll still need to handle creating a unique name for the secret given we can't avoid users creating secrets with specific names.

Not related to this PR, but we may eventually want to support people using an existing secret for values rather than entering values.